client/packrat-ui/form/index.js

```diff
@@ -66,42 +86,73 @@ const ReusableForm = forwardRef((props, ref) => {
     focus: focusInput,
   }));
 
+  useEffect(() => {
+    const subscription = watch((data) => {
+      console.log('data', data);
+      try {
+        schema.parse(data);
+        clearErrors();
+        setIsValidated(false);
+      } catch (error) {
+        // console.log('here is error', error);
+        trigger();
+      }
+    });
+    return () => subscription.unsubscribe();
+  }, [watch]);
+
   return (
-    <Form onSubmit={handleSubmit(onSubmit)}>
+    <Form onSubmit={() => handleSubmit(onSubmit)}>
       {fields.map((field) => (
-        <XStack
-          overflow="hidden"
-          space="$2"
-          margin="$3"
-          padding="$2"
-          key={field.name}
-        >
-          {field.label && <Label htmlFor={field.name}>{field.label}</Label>}
+        <XStack overflow="hidden" key={field.name} flexDirection="column">
+          {field.label && (
+            <Label
+              htmlFor={field.name}
+              color="gray"
+              fontSize={15}
+              _dark={{
+                color: 'warmGray.200',
+              }}
+            >
+              {field.label}
+            </Label>
+          )}
           <YStack space="$1">
             <Controller
```

Comment:
if we use inputtext component we have double controllers, we should probably just use a tamagui input

Reply:
i think this is why its so laggy if you keep changing inputs and trying to submit

Reply:
I thought i had a slow pc